From 2cfa06532ce9d59b9b8fd45da10e1cdb8b72ac78 Mon Sep 17 00:00:00 2001
From: Xiaofeng Wei <xiaofeng.wei@nxp.com>
Date: Tue, 11 Mar 2025 10:21:15 +0800
Subject: [PATCH 26/30] arm64: dts: imx8mp-frdm: add AP1302 support

Add new dts file to support AP1302/AR0144 for i.MX8MP FRDM platform.

Signed-off-by: Xiaofeng Wei <xiaofeng.wei@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |   3 +-
 .../boot/dts/freescale/imx8mp-frdm-ap1302.dts | 125 ++++++++++++++++++
 2 files changed, 127 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-frdm-ap1302.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index ad38e915878e..238babc8fb54 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -142,7 +142,8 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mp-ddr4-evk.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-evk-ndm.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm.dtb imx8mp-frdm-os08a20.dtb imx8mp-frdm-dual-os08a20.dtb \
-			  imx8mp-frdm-boe-wxga-lvds0-panel.dtb imx8mp-frdm-boe-wxga-lvds1-panel.dtb
+			  imx8mp-frdm-boe-wxga-lvds0-panel.dtb imx8mp-frdm-boe-wxga-lvds1-panel.dtb \
+			  imx8mp-frdm-ap1302.dtb
 
 imx8mp-evk-revb4-dtbs := imx8mp-evk.dtb imx8mp-evk-revb4.dtbo
 imx8mp-evk-revb4-rm67191-dtbs := imx8mp-evk-rm67191.dtb imx8mp-evk-revb4.dtbo
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-frdm-ap1302.dts b/arch/arm64/boot/dts/freescale/imx8mp-frdm-ap1302.dts
new file mode 100644
index 000000000000..9a0961b147e7
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-frdm-ap1302.dts
@@ -0,0 +1,125 @@
+/*
+* Copyright 2025 NXP
+* SPDX-License-Identifier: (GPL-2.0-only OR MIT)
+*/
+
+/dts-v1/;
+
+#include "imx8mp-frdm.dts"
+
+/ {
+	reg_dvdd_sel: regulator-dvdd_sel {
+		compatible = "regulator-fixed";
+		regulator-name = "DVDD_SEL";
+		gpio = <&adp5585gpio_isp 0 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		startup-delay-us = <2000>;
+	};
+
+	reg_dvdd_1v2: regulator-dvdd {
+		compatible = "regulator-fixed";
+		regulator-name = "DVDD_1V2";
+		gpio = <&adp5585gpio_isp 6 GPIO_ACTIVE_HIGH>;
+		regulator-min-microvolt = <1200000>;
+		regulator-max-microvolt = <1200000>;
+		enable-active-high;
+		vin-supply = <&reg_dvdd_sel>;
+	};
+
+	reg_vdd_3v3: regulator-vdd {
+		compatible = "regulator-fixed";
+		regulator-name = "VDD_3V3";
+		gpio = <&adp5585gpio_isp 5 GPIO_ACTIVE_HIGH>;
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		startup-delay-us = <4000>;
+		enable-active-high;
+	};
+
+	reg_vddio_1v8: regulator-vddo {
+		compatible = "regulator-fixed";
+		regulator-name = "VDDIO_1V8";
+		gpio = <&adp5585gpio_isp 9 GPIO_ACTIVE_HIGH>;
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		startup-delay-us = <4000>;
+		enable-active-high;
+		vin-supply = <&reg_vdd_3v3>;
+	};
+
+	reg_vaa_sel: regulator-vaa_sel {
+		compatible = "regulator-fixed";
+		regulator-name = "VAA_SEL";
+		gpio = <&adp5585gpio_isp 1 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
+	reg_avdd_2v8: regulator-avdd {
+		compatible = "regulator-fixed";
+		regulator-name = "AVDD_2V8";
+		gpio = <&adp5585gpio_isp 7 GPIO_ACTIVE_HIGH>;
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		enable-active-high;
+		vin-supply = <&reg_vaa_sel>;
+	};
+};
+
+&i2c2 {
+	/delete-node/ov5640_mipi@3c;
+
+	adp5585_isp: mfd-isp@34 {
+		compatible = "adi,adp5585";
+		reg = <0x34>;
+		status = "okay";
+
+		adp5585gpio_isp: gpio-isp@34 {
+			compatible = "adp5585-gpio";
+			gpio-controller;
+			#gpio-cells = <2>;
+		};
+
+		adp5585pwm_isp: pwm-isp@34 {
+			compatible = "adp5585-pwm";
+			#pwm-cells = <3>;
+		};
+	};
+
+	ap1302: ap1302_mipi@3c {
+		compatible = "onsemi,ap1302";
+		reg = <0x3c>;
+		csi_id = <0>;
+		reset-gpios = <&pcal6416 0 GPIO_ACTIVE_LOW>;
+		isp_en-gpios  = <&adp5585gpio_isp 2 GPIO_ACTIVE_HIGH>;
+		DVDD-supply   = <&reg_dvdd_1v2>;
+		VDDIO-supply  = <&reg_vddio_1v8>;
+		AVDD-supply   = <&reg_avdd_2v8>;
+		status = "okay";
+
+		port {
+			ap1302_mipi_0_ep: endpoint {
+				data-lanes = <1 2>;
+				clock-lanes = <0>;
+				remote-endpoint = <&mipi_csi0_ep>;
+			};
+		};
+	};
+};
+
+
+&mipi_csi_0 {
+	status = "okay";
+
+	/delete-node/port;
+	port {
+		mipi_csi0_ep: endpoint {
+			remote-endpoint = <&ap1302_mipi_0_ep>;
+			data-lanes = <2>;
+			csis-hs-settle = <9>;
+		};
+	};
+};
+
+&mipi_csi_1 {
+	status = "disabled";
+};
-- 
2.34.1

